### PR TITLE
New matrix table get interface.

### PR DESCRIPTION
**Pull request by huxuan**
- Use 1-d array instead of 2-d array.
- Use raw c types instead of c++ ones.

Similar to #59 .

### CODE REVIEW

File: src/table/matrix_table.cpp [hunk 1]
```diff
@@ -90,6 +90,19 @@ void MatrixWorkerTable<T>::Get(const std::vector<integer_t>& row_ids,
 }
 
 template <typename T>
+void MatrixWorkerTable<T>::Get(T* data, size_t size, integer_t* row_ids,
```
Comment (you-n-g) [inline src/table/matrix_table.cpp]: It's better that the name of variable `size` changed to `col_size`.
I'm also confused why this variable is needed if it must be num_col_.  @feiga

Comment (huxuan) [inline src/table/matrix_table.cpp]: Done for rename.

Also doubt about the necessity of `size` or `col_size`. Maybe it's just for checking?

Comment (feiga) [inline src/table/matrix_table.cpp]: No, I think the size should be the size of the 'data' array. This is implemented by you or Xiao? You need to make sure with him.

Comment (feiga) [inline src/table/matrix_table.cpp]: You may take a look at other Get function I implemented before. The size means differently

Comment (huxuan) [inline src/table/matrix_table.cpp]: @feiga This function is implemented by me by referring to the one above. In that function the `size`   is checked with `num_col_` [1].

[1] https://github.com/Microsoft/multiverso/pull/62/files/225562384d692162659131496f99e15ac2b1e0c9#diff-fa4535afda5d4316b5601c83a8de7cc0R82

Comment (feiga) [inline src/table/matrix_table.cpp]: In that function, the size means the size of each array in ```data_vec```. I think your function should be the size of data array, which is  num_col * row_ids_size

Comment (huxuan) [inline src/table/matrix_table.cpp]: OK, got then. So the answer for why we need the `size` is for check only? I wonder if there is any other purpose.

Comment (feiga) [inline src/table/matrix_table.cpp]: Yes. The main consideration is that when user pass an array as argument, he need to pass both the pointer and the array size.

Comment (huxuan) [inline src/table/matrix_table.cpp]: OK, this convince me. So what do you think of then, @you-n-g ?

Comment (you-n-g) [inline src/table/matrix_table.cpp]: I'm convinced too.

Comment (huxuan) [inline src/table/matrix_table.cpp]: The PR has been updated as expected. Would you please have another review?

Comment (you-n-g) [inline src/table/matrix_table.cpp]: I think it can be merged now.

**Pull request merged**